### ISSUE TITLE
Create string context helper for paths and use in appropriate places

### ISSUE THREAD

**Issue by mightybyte** [haskell-nix/hnix#424]
See also https://github.com/haskell-nix/hnix/pull/405/files/95d4b9406d80d9f9e86ce0125a0025d990ab8448#r240045704

**Comment by jwiegley** [comment]
@mightybyte Hi Doug, can you elaborate on this issue further? Since it's marked "Good First Issue", it should contain information for anyone to get started quickly, and to know when they've done enough to consider it finished. Thanks!

**Comment by sjakobi** [comment]
I agree that it would be good to have a bit more info on what should be done here.
